### ISSUE TITLE
get parents

### ISSUE THREAD

**Issue by pnrobinson** [Phenomics/ontolib#18]
Hi!
Are we planning to offer methods such as getParents() ? (distinct from getAncestors?)
It would be useful to have this as a simple API function.

**Comment by holtgrewe** [comment]
Agreed, is currently only available through graph property of `MinimalOntology<>` and should be a convenience function in `Ontology<>`.

**Comment by holtgrewe** [comment]
Should be fine after merging #20.

**Comment by iamkoehler** [comment]
Does` getParents(Term t)` and `getAncestors(Term t)` return the `Term t` itself?  This is an important design decision. In ontologizer it did.

**Comment by holtgrewe** [comment]
`getParents()` does not return the parents, a term is not its own parent.

As for `getAncestors()`, there is a `boolean` flag and by default, it returns itself, a term is contained in the set of ancestors.

cf. http://javadoc.io/doc/com.github.phenomics/ontolib-core/0.3

This might be a break of consistency and principle of least surprise.

Maybe `getParents()` should not return itself, as should `getAncestors()` and then there should be a `getParentsAndSelf()` and a `getAncestorsAndSelf()`?

**Comment by iamkoehler** [comment]
I think this is a good solution.
